### DISCUSSION TITLE
Notifications spam fix (LSP progress)

### DISCUSSION THREAD

**Discussion by smileart** [folke/noice.nvim#837] [Q&A]
Trying to suppress the noisy notifications (shown on every keystroke) that produced by `harper_ls`[[*](https://github.com/elijah-potter/harper)] I followed the respective [instructions](https://github.com/folke/noice.nvim/wiki/Configuration-Recipes#ignore-certain-lsp-servers-for-progress-messages), but to no avail. Perhaps despite the "LSP Message" title the type is not in fact `lsp`? How do I find out which type/kind to filter for a particular lsp/plugin? Any help would be appreciated. 🙏 

<img width="715" alt="image" src="https://github.com/folke/noice.nvim/assets/455338/f9aeb816-e109-4b68-8393-269826f0a486">

```lua
      {
        filter = {
          event = "lsp",
          kind = "progress",
          cond = function(message)
            local client = vim.tbl_get(message.opts, "progress", "client")
            return client == "harper_ls"
          end,
        },
        opts = { skip = true },
      }
```

P.S. Using lazy.vim-based setup with `noise.nvim` and `nvim-notify`.

**Comment by ilias777** [comment]
These are LSP messages. I think you can't use here a filter to skip these messages.
You can read here &rarr; [LSP Messages](https://github.com/folke/noice.nvim/wiki/A-Guide-to-Messages#lsp-messages)

I don't use `harper_ls`, but you can try to change `value.title ==` to `Linting` like so:

```lua
local harper_ls_token = nil

if name == 'harper_ls' then
    if result.token == harper_ls_token then
        return
    end
    if value.title == 'Linting' then
        harper_ls_token = result.token
        return
    end
end
```

Maybe this helps.

**Comment by smileart** [comment]
Thanks for the answer, @ilias777.

I've tried everything, debugged the titles and tokens in this handler, but couldn't intercept the pesky `harper_ls` in any way. Luckily, I've updated the language server via Mason and the annoying notifications disappeared. I've checked their repo, but at a glance couldn't see any recent Issues / PRs addressing this particular issue. Either way, the problem has gone, but I believe the doc you referenced could be useful in the future and maybe this thread would improve SEO for finding it. Cheers. 🤝
